script.rb

```diff
@@ -0,0 +1,184 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    length.times do
+      yield self[i]
+      i += 1
+    end
+  end
```

Comment:
- [ ] your method should return self after calling yield for each element in the self. That is how the original each method behaves but yours does not return `self`. Your my_each method should be able to work on an array, range, and hash just like the original each method. The best way to make sure your method works like the original method is to test it out against the original method.
`[1, 2, 3, 4, 5].my_each {|v| p v * 2}` should behave exactly as  `[1, 2, 3, 4, 5].each {|v| p v * 2}`
This should apply for when self is a range or hash.